vminitd/Sources/vminitd/ManagedProcess.swift

```diff
@@ -143,131 +149,151 @@ final class ManagedProcess: Sendable {
         self.command = command
         self.terminal = stdio.terminal
         self.bundle = bundle
-        self.state = Mutex(State(io: io))
+        self.state = Mutex(State(io: io, errorPipe: errorPipe))
     }
 }
 
 extension ManagedProcess {
     func start() throws -> Int32 {
-        try self.state.withLock {
-            log.info(
-                "starting managed process",
-                metadata: [
-                    "id": "\(id)"
-                ])
-
-            // Start the underlying process.
-            try command.start()
-            defer {
-                try? self.ackPipe.fileHandleForWriting.close()
-                try? self.syncPipe.fileHandleForReading.close()
-                try? self.ackPipe.fileHandleForReading.close()
-                try? self.syncPipe.fileHandleForWriting.close()
-            }
-
-            // Close our side of any pipes.
-            try $0.io.closeAfterExec()
-            try self.ackPipe.fileHandleForReading.close()
-            try self.syncPipe.fileHandleForWriting.close()
+        do {
+            return try self.state.withLock {
+                log.info(
+                    "starting managed process",
+                    metadata: [
+                        "id": "\(id)"
+                    ])
 
-            let size = MemoryLayout<Int32>.size
-            guard let piddata = try syncPipe.fileHandleForReading.read(upToCount: size) else {
-                throw ContainerizationError(.internalError, message: "no PID data from sync pipe")
-            }
+                // Start the underlying process.
+                try command.start()
 
-            guard piddata.count == size else {
-                throw ContainerizationError(.internalError, message: "invalid payload")
-            }
+                defer {
+                    try? self.ackPipe.fileHandleForWriting.close()
```

Comment:
Related to my other latest comment, if we just have errorPipe as a variable on ManagedProcess (not in the state mutex) add a close for it here.